src/client/wire_protocol/flags.rs

```diff
@@ -0,0 +1,108 @@
+pub struct OpReplyFlags {
+    cursor_not_found: bool, // Bit 0
+    query_failure: bool,    // Bit 1
+    await_capable: bool,    // Bit 3
+
+    // All bits remaining must be 0
+}
+
+impl OpReplyFlags {
+    pub fn from_i32(i: i32) -> OpReplyFlags {
+        let cursor_not_found = (i & 1) != 0;
+        let query_failure = (i & (1 << 1)) != 0;
+        let await_capable = (i & (1 << 3)) != 0;
+
+        OpReplyFlags { cursor_not_found: cursor_not_found,
+                       query_failure: query_failure,
+                       await_capable: await_capable }
+    }
+}
+
+pub struct OpInsertFlags {
+    pub continue_on_error: bool,  // Bit 0
+
+    // All bits remaining must be 0
+}
+
+pub struct OpQueryFlags {
+    pub tailable_cursor: bool,    // Bit 1
+    pub slave_ok: bool,           // Bit 2
+    pub oplog_relay: bool,        // Bit 3
+    pub no_cursor_timeout: bool,  // Bit 4
+    pub await_data: bool,         // Bit 5
+    pub exhaust: bool,            // Bit 6
+    pub partial: bool,            // Bit 7
+
+    // All bits remaining must be 0
+}
+
+
+impl OpInsertFlags {
+    pub fn no_flags() -> OpInsertFlags {
+        OpInsertFlags { continue_on_error: false }
+    }
+
+    pub fn to_i32(&self) -> i32 {
+        if self.continue_on_error {
+            1
+        } else {
+            0
+        }
+    }
+}
+
+impl OpQueryFlags {
+    pub fn no_flags() -> OpQueryFlags {
+        OpQueryFlags { tailable_cursor: false, slave_ok: false,
+                       oplog_relay: false, no_cursor_timeout: false,
+                       await_data: false, exhaust: false, partial: false }
+    }
+
+    pub fn to_i32(&self) -> i32 {
```

Comment:
Hmmm....I'm not actually sure how I ended up doing that in the way I did, but you're right, it's definitely unnecessary to have the temporary variable.